tests/click_tests/device_menu/test_wipe_code.py

```diff
@@ -0,0 +1,312 @@
+    _input_see_confirm,
+    _see_pin,
+)
+from .common import (
+    Menu,
+    assert_device_screen,
+    close_device_menu,
+    enter_pin,
+    open_device_menu,
+)
+from .test_pin import Situation, prepare_pin_dialogue
+
+if TYPE_CHECKING:
+    from trezorlib.debuglink import DebugLink
+
+    from ...device_handler import BackgroundDeviceHandler
+# Trezor Safe 7 only
+pytestmark = [pytest.mark.models("eckhart")]
+
+WIPE_CODE = "2"
+WIPE_CODE_TITLE = "wipe_code__title"
+
+
+def change_wipe_code(
+    debug: "DebugLink", pin: str = PIN4, wipe_code: str = WIPE_CODE
+) -> None:
+    # Intro screen
+    go_next(debug)
+    # Enter pin to authorize
+    _input_see_confirm(debug, pin)
+    # Enter new wipe code twice
+    _enter_two_times(debug, wipe_code, wipe_code)
+    # Confirmation screen
+    go_next(debug)
+
+
+def prepare_wipe_dialogue(
+    device_handler: "BackgroundDeviceHandler", situation: Situation
+) -> None:
+
+    debug = device_handler.debuglink()
+    features = device_handler.features()
+    wipe_title = TR.translate(WIPE_CODE_TITLE)
+    assert wipe_title in Menu.SECURITY.content(features)
+
+    if situation in (Situation.REMOVE, Situation.CHANGE):
+        assert features.wipe_code_protection is False
+        open_device_menu(debug)
+        Menu.SECURITY.navigate_to(debug, features)
+        layout = debug.read_layout()
+        idx = layout.vertical_menu_content().index(TR.translate(WIPE_CODE_TITLE))
+        debug.button_actions.navigate_to_menu_item(idx)
+
+        change_wipe_code(debug)
+
+        assert_device_screen(debug, Menu.SECURITY)
+        close_device_menu(debug)
+
+        features = device_handler.features()
+        assert features.wipe_code_protection is True
+
+    # Open device menu
+    open_device_menu(debug)
+
+    assert features.pin_protection is True
+
+    if situation is Situation.SETUP:
+        assert features.wipe_code_protection is False
+        # Wipe code submenu shouldn't be present
+        assert Menu.WIPE_CODE.content(features) is None
+        # Navigate to security menu
+        Menu.SECURITY.navigate_to(debug, features)
+        title = TR.translate(WIPE_CODE_TITLE)
+    else:
+        assert features.wipe_code_protection is True
+        # Navigate to wipe code menu
+        Menu.WIPE_CODE.navigate_to(debug, features)
+
+        if situation is Situation.CHANGE:
+            title = TR.wipe_code__change
+
+        elif situation is Situation.REMOVE:
+            title = TR.wipe_code__remove
+        else:
+            raise RuntimeError("Unknown situation")
+
+    # Trigger wipe code dialogue
+    layout = debug.read_layout()
+    idx = layout.vertical_menu_content().index(title)
+    debug.button_actions.navigate_to_menu_item(idx)
+
+
+@pytest.mark.setup_client(pin=None)
+def test_wipe_code_setup_pin_unset(device_handler: "BackgroundDeviceHandler"):
+
+    # Device doesn't have the pin set, the wipe code button is not accessible
+    with pytest.raises(
+        AssertionError,
```

Comment:
nit: can we use a specific exception type for this error?
(i.e. a new exception type that is raised when the wipe code menu is unavailable)